03_letters/brvishnu92/main_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := `a:2
+b:1
+`
+	actual := buf.String()
+	assert.Equalf(t, expected, actual, "got %v want %v", actual, expected)
+}
+
+func TestLetters(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"test1":   {input: "blahbla", expected: map[rune]int{97: 2, 98: 2, 104: 1, 108: 2}},
+		"test2":   {input: "smarty", expected: map[rune]int{97: 1, 109: 1, 114: 1, 115: 1, 116: 1, 121: 1}},
+		"empty":   {input: "", expected: map[rune]int{}},
+		"sameval": {input: "aaaa", expected: map[rune]int{97: 4}},
+		"oneval":  {input: "a", expected: map[rune]int{97: 1}},
+	}
+	for name, test := range testCases {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			result := letters(test.input)
+			assert.Equalf(t, test.expected, result, "got %v want %v", result, test.expected)
+		})
+	}
+}
+
+func TestSortLetters(t *testing.T) {
+	testCases := map[string]struct {
+		expected []string
+		input    map[rune]int
+	}{
+		"test1":   {expected: []string{"a:2", "b:2", "h:1", "l:2"}, input: map[rune]int{97: 2, 98: 2, 104: 1, 108: 2}},
+		"empty":   {expected: []string{}, input: map[rune]int{}},
+		"sameval": {expected: []string{"a:4"}, input: map[rune]int{97: 4}},
+		"oneval":  {expected: []string{"a:1"}, input: map[rune]int{97: 1}},
```

Comment:
"sameval" and "oneval" are really the same test in that the code does the same thing in both cases. You only need one of them.

Reply:
Yeah, I agree. Have removed the test case.